src/snowflake/snowpark/modin/plugin/extensions/dataframe_overrides.py

```diff
@@ -1065,6 +1238,9 @@ def insert(
     # Dictionary keys are treated as index column and this should be joined with
     # index of target dataframe. This behavior is similar to 'value' being DataFrame
     # or Series, so we simply create Series from dict data here.
+    if isinstance(value, set):
+        raise TypeError(f"'{type(value).__name__}' type is unordered")
```

Comment:
is this new behavior, and do we have a test for it?

Reply:
Not new behavior but yes, there is a test for this. We just didn't have this check in place before